### PR TITLE
[CORE] Minor refactor of use ValidationResult for supportWriteFilesExec

### PR DESCRIPTION
**Pull request by Yohahaha**

### CODE REVIEW

**Comment by Yohahaha** [comment]
@ulysses-you @JkSelf could you help rerun failed CI jobs and review it? thanks!

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala [hunk 1]
```diff
@@ -140,22 +140,22 @@ object BackendSettings extends BackendSettingsApi {
...
-      options: Map[String, String]): Option[String] = {
+      options: Map[String, String]): ValidationResult = {
 
     def validateCompressionCodec(): Option[String] = {
       // Velox doesn't support brotli and lzo.
       val unSupportedCompressions = Set("brotli, lzo")
       val compressionCodec = WriteFilesExecTransformer.getCompressionCodec(options)
       if (unSupportedCompressions.contains(compressionCodec)) {
-        Some("brotli or lzo compression codec is not support in velox backend.")
+        Some("brotli or lzo compression codec is not support in Velox backend.")
```
Comment (JkSelf) [inline backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala:150]: `brotli`  -> `Brotli`   and `support`  -> `supported`

**Review by JkSelf — COMMENTED** [review]
Good catch. LGTM except one small comment.

File: backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala (hunk 1 above)
Comment (Yohahaha) [reply backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala]: ok.

**Review by ulysses-you — APPROVED** [review]

**Pull request merged**